vcr/matchers.py

```diff
@@ -30,12 +32,29 @@ def query(r1, r2):
     return r1.query == r2.query
 
 
-def body(r1, r2):
+def raw_body(r1, r2):
     if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
         return r1.body.read() == r2.body.read()
     return r1.body == r2.body
 
 
+def body(r1, r2):
+    if hasattr(r1.body, 'read') and hasattr(r2.body, 'read'):
+        r1_body = r1.body.read()
+        r2_body  = r2.body.read()
+    else:
+        r1_body = r1.body
+        r2_body  = r2.body
+    if r1.headers.get('Content-Type') == r2.headers.get('Content-Type') == 'application/x-www-form-urlencoded':
```

Comment:
I think leaving the header matcher as is is fine for now. We could maybe add another one that specifically indicates that the matching is case insensitive, but keep in mind that this matching should be happening on requests that are generated from the same test code so there probably shouldn't be differences in the casing the requst headers.

One edge case we might (or might not) want to think about is the case where two headers with different casing are sent with the request e.g.

'Content-Type' and 'content-type'. The behavior of CaseInsensitveDict is undefined in this case.